### PR TITLE
Refactor how network state is passed to transaction controller

### PR DESCRIPTION
**Pull request by Gudahtt**
The network state is now passed to the TransactionController via a getter function and a subscription function, instead of passing one of the network controller stores directly.

This way of passing the state makes further refactoring easier, as we don't have to change the input when the store is changed or replaced. It's also more aligned with our conventions today.

This change was made as part of https://github.com/MetaMask/controllers/issues/971.

## Manual Testing Steps

There should be zero functional changes.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by brad-decker — APPROVED** [review]
LGTM

File: app/scripts/controllers/transactions/tx-state-manager.js [hunk 1]
```diff
@@ -81,7 +86,7 @@ export default class TransactionStateManager extends EventEmitter {
    * @returns {TransactionMeta} the default txMeta object
    */
   generateTxMeta(opts = {}) {
-    const netId = this.getNetwork();
+    const netId = this.getNetworkState();
```
Comment (mcmire) [inline app/scripts/controllers/transactions/tx-state-manager.js:89]: Side note, we should make these variables consistent at some point too. It's confusing that sometimes this is `netId`, sometimes, `network`, sometimes `networkState`.

**Review by mcmire — APPROVED** [review]
LGTM.

File: app/scripts/controllers/transactions/tx-state-manager.js (hunk 1 above)
Comment (Gudahtt) [reply app/scripts/controllers/transactions/tx-state-manager.js:89]: The value itself is confusing, even apart from the name. It's a mix of "loading" and network ID. We use it in some places _calling it_ "network ID", but using it just for the "loading" part.

Clarifying this is on the API normalization list.

**Pull request merged**